src/Metadata/Metadata.php

```diff
@@ -104,11 +105,17 @@ public function withShortName(string $shortName): static
         return $self;
     }
 
+    /**
+     * @return class-string|null
```

Comment:
This is the main change, everything else is changing `'foo'` by `\stdClass::class` in tests to avoid PHPStan failure.